src/errors.rs

```diff
@@ -88,7 +89,14 @@ pub enum RustupError {
     ToolchainNotInstallable(String),
     #[error("toolchain '{0}' is not installed")]
     ToolchainNotInstalled(String),
-    #[error("no override and no default toolchain set")]
+    #[error("{} have not specified a toolchain version to use.\n{}\n{}",
+        match utils::current_name() {
+            Some(arg0) if arg0.starts_with("rustup") => "you".to_string(),
```

Comment:
I think language wise, that rather than assigning motive and will to the process that ends up in this state, just stating the state will be less acrimonious:

'Rustup cannot choose a rust version because there is no default toolchain, and no override has been provided.' rather than 'you have not specified a toolchain version to use.'  What if its a CI bot encountering this, or if a user *has* specified a version and now we're arguing with them. Better to avoid that.

Reply:
I've intentionally gone with active voice here, because that makes it clear the user is expected to do something. "There is no default toolchain" makes it sound like there was supposed to be one, but in reality it's up to the user to explicitly make that decision. Do you have any specific use case in mind when a user has specified a version, but we still hit this error?

Reply:
It sounds like the problem that occurred was a very quite error stuck deep in a set of build scripts. The user didn't even locate the error for some time as I understand it.

If a user has *actually* set a version then we won't hit this: I'm talking about the experience the user will have if they think they have set a version and have things configured correctly.

Consider discussions / errors like this https://www.reddit.com/r/rust/comments/acapjb/rust_error_messages_are_so_quite_3/

The diagnostic is passive. The call to action is active. Thats what I'm proposing here.

Reply:
The user located the error immediately, they just didn't attribute it to rustup. 

How about something like this then:

```
error: rustup could not choose a version of [$0] to run, because it wasn't specified explicitly, and no default is configured.
help: run 'rustup default stable' to download the latest stable release of Rust and set it as default.
         See https://rust-lang.github.io/rustup/overrides.html for more ways to specify the Rust version for your project.
```

Reply:
I actually really like the "for your project" phrasing, because it should make it clear to the user that they shouldn't need to read that doc if they're not building their own project.

Reply:
My experience with folk working through a series of problems happening to something they haven't fully come to grips with is that too much information leads to cognitive shortcuts being used. 

In this case, giving the user an option to run a command (most sensible thing to do), or visit a webpage and read about a much more complicated thing that most people don't need and shouldn't use, is not helpful to the user. The error message doesn't need to explain all the ways that rustup not being able to find a rust install can happen, nor conversely all the possible solutions for that situation. I don't think it even needs to explain the top-two: it needs to provide a sensible conservative solution with little to no adverse side effects, and users can consider other options later.

Reply:
Thinking about it more, I can't really think of a situation where a developer encounters this error and isn't already familiar with rustup, so I actually agree with you here - we can tailor the message to not-(quite)-technical users.